### PR TITLE
Add Gitlab support

### PR DESCRIPTION
**Pull request by thaligar**
Hi, I added GitLab support. I tested it with online and diy-GitLab instance.
I realized the polymorphism via enums.
Please give me a buzz if there is anything not according to your standards or if you prefer **'dyn traits'** over **enums**.
I am happy to change to your needs.

### CODE REVIEW

**Comment by thaligar** [comment]
Fixed the failed test and the formatting.

**Comment by thaligar** [comment]
Fixed the clippy issues.

**Comment by thanethomson** [comment]
Hey @thaligar! Sorry for the delay in getting feedback to you here, I've been swamped with other work over the past couple of weeks.

Thanks for your thoughtful contribution here! I've fixed the latest clippy warnings/errors for Rust v1.67 in #42 - please rebase your PR here and I'll review this soon!

**Comment by thaligar** [comment]
Hi @thanethomson, no problem I thought so. 
I rebased the PR as asked.

File: src/vcs.rs [hunk 1]
```diff
@@ -25,6 +25,19 @@ impl PlatformId {
     }
 }
 
+/// Generic definition of an online Git project.
+pub trait GitProject {
```
Comment (thanethomson) [inline src/vcs.rs:29]: I'd recommend rather calling this `GenericProject`, and renaming the `OnlineGitProject` enum to `Project`.

Both are members of the `vcs` package, so I think it's clear that they're VCS projects.

File: .changelog/unreleased/features/76-add-gitlab-support.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+- Add support for GitLab repositories.
+  ([#76](https://github.com/informalsystems/unclog/pull/76))
```
[the remark this answers is not in the document]
Comment (thaligar) [reply .changelog/unreleased/features/76-add-gitlab-support.md:2]: Right, will correct.

File: src/error.rs [hunk 3]
```diff
@@ -82,4 +82,6 @@ pub enum Error {
     InvalidUrl(String),
     #[error("component \"{0}\" is not defined in changelog config.toml file")]
     ComponentNotDefined(String),
+    #[error("project type could not be deduced from url \"{0}\"")]
+    ProjectCouldNotBeDeduced(String),
```
[the remark this answers is not in the document]
Comment (thaligar) [reply src/error.rs:86]: UnrecognizedProjectType does already exist in line 44. I would then suggest to use the already existing message instead.

**Comment by thaligar** [comment]
Hi @thanethomson, fixed all issues. I added a commit at the "UnrecognizedProjectType". Fmt, clippy and tests were run before pushing.

**Review by thanethomson — APPROVED** [review]
Awesome, thanks very much for this @thaligar!

**Comment by thaligar** [comment]
@thanethomson No problem. Thanks for the great tool.

**Pull request merged**